### PR TITLE
Support DOT syntax for getChildByName

### PR DESCRIPTION
**Pull request by zii**
When I load GUI from Cocos Studio directly, it is not very easy to get deeply widget.

So, how about this:
layer->getChildByName("root.panel.button")
layer->removeChildByName("root.panel.button")

### CODE REVIEW

**Comment by cmzx3444** [comment]
Nice feature!

File: cocos/2d/CCNode.cpp [hunk 1]
```diff
@@ -821,6 +821,16 @@ Node* Node::getChildByTag(int tag) const
...
+
+    int dotpos = name.find('.');
+    if (dotpos > 0)
+    {
+        first_name = name.substr(0, dotpos);
+        suffix = name.substr(dotpos+1);
+    }
     
     std::hash<std::string> h;
     size_t hash = h(name);
```
Comment (cmzx3444) [inline cocos/2d/CCNode.cpp]: size_t hash = h(name); 
the name should be first_name

File: cocos/2d/CCNode.cpp [hunk 2]
```diff
@@ -829,7 +839,7 @@ Node* Node::getChildByName(const std::string& name) const
     {
         // Different strings may have the same hash code, but can use it to compare first for speed
         if(child->_hashOfName == hash && child->_name.compare(name) == 0)
```
Comment (cmzx3444) [inline cocos/2d/CCNode.cpp]: child->_name.compare(name) 
the name should be first_name

**Comment by zii** [comment]
Thank you very much! I nearly make a big bug.

**Comment by zii** [comment]
Is there faster implemention?

**Comment by minggo** [comment]
@zii 
Thanks. But it will break the compatibility.
I think it is better to add a helper function for it.

**Comment by zii** [comment]
You are right sir, it is a little complicated. Suite for script level.

**Pull request closed without merging**